base/strings/search.jl

```diff
@@ -127,16 +127,17 @@ findfirst(ch::AbstractChar, string::AbstractString) = findfirst(==(ch), string)
 function findnext(testf::Function, s::AbstractString, i::Integer)
     i = Int(i)
     z = ncodeunits(s) + 1
-    1 ≤ i ≤ z || throw(BoundsError(s, i))
+    1 ≤ i ≤ z || throw(BoundsError(s, i))
     @inbounds i == z || isvalid(s, i) || string_index_err(s, i)
-    for (j, d) in pairs(SubString(s, i))
-        if testf(d)
-            return i + j - 1
-        end
+    e = lastindex(s)
+    @inbounds while i <= e
+        testf(s[i]) && return i
```

Comment:
This might run `testf` with bounds check off, it was [suggested](https://github.com/JuliaLang/julia/pull/34885#discussion_r384491202) to me to not do that.